packages/overlay/src/vaadin-overlay-position-mixin.js

```diff
@@ -262,7 +274,8 @@ export const PositionMixin = (superClass) =>
     __shouldAlignStartVertically(targetRect) {
       // Using previous size to fix a case where window resize may cause the overlay to be squeezed
       // smaller than its current space before the fit-calculations.
-      const contentHeight = Math.max(this.__oldContentHeight || 0, this.$.overlay.offsetHeight);
+      const contentHeight =
+        this.requiredVerticalSpace || Math.max(this.__oldContentHeight || 0, this.$.overlay.offsetHeight);
```

Comment:
There are upsides and downsides to this approach. If the overlay content is smaller than `requiredVerticalSpace` it will still open to the top, which might be inconsistent with other components. On the other hand, having a consistent open direction while changing the content, like filtering items in the combo box, is probably more desirable. I would suggest to double check this new behavior with either Rolf or Jouni.

Reply:
Yeah, couldn't come up with a better solution after some deep thinking, so let's go with this behavior.